tests/integration/links/test_journal.py

```
@@ -96,9 +96,28 @@ def test_a_gap_survives_the_goodbye_that_closes_the_run(tmp_path: Path):
         assert record.gapped
 
     # The disk that lost the event is the disk the gap marker is written to, so
-    # both go at once. What is left must not read back as a run to resume.
+    # both can fail at once. The log must not grow over a record that has not
+    # admitted the gap: a short log is a cast to pick up, a holed one is not.
     @staticmethod
-    def test_a_gap_that_cannot_be_written_takes_the_record_with_it(tmp_path: Path):
+    def test_an_unwritable_gap_refuses_the_next_append(tmp_path: Path):
```

Ruff (0.16.2)

[warning] Missing return type annotation for staticmethod `test_an_unwritable_gap_refuses_the_next_append`

Add return type annotation: `None`

(ANN205)